pkg/apis/apps/validation/validation_test.go

```diff
@@ -2248,6 +2248,30 @@ func TestValidateDaemonSetUpdate(t *testing.T) {
 				},
 			},
 		},
+		"Read-write volume verification": {
+			old: apps.DaemonSet{
+				ObjectMeta: metav1.ObjectMeta{Name: "abc", Namespace: metav1.NamespaceDefault},
+				Spec: apps.DaemonSetSpec{
+					Selector:           &metav1.LabelSelector{MatchLabels: validSelector},
+					TemplateGeneration: 1,
+					Template:           validPodTemplateAbc.Template,
+					UpdateStrategy: apps.DaemonSetUpdateStrategy{
+						Type: apps.OnDeleteDaemonSetStrategyType,
+					},
+				},
+			},
+			update: apps.DaemonSet{
+				ObjectMeta: metav1.ObjectMeta{Name: "abc", Namespace: metav1.NamespaceDefault},
+				Spec: apps.DaemonSetSpec{
+					Selector:           &metav1.LabelSelector{MatchLabels: validSelector},
+					TemplateGeneration: 2,
+					Template:           readWriteVolumePodTemplate.Template,
```

Comment:
To test this, you will need to set the gate in the test loop, but you need to test that it fails with the gate off and passes with the gate on, which means adding a field to the testcase struct and having one success case and one fail case.  The new field can default to false, so only the cases that need to set it specify the field at all.

Look for examples in this same file, e.g.

```
 defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.StatefulSetAutoDeletePVC, true)()
```

combined with `t.Run()` is sets the gate, runs the case, then un-sets the gate.